pkg/migrations/seq_num.go

```diff
@@ -1,30 +1,23 @@
 package migrations
 
 import (
-	"github.com/pkg/errors"
 	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/pkg/migrations/internal"
-	"github.com/stackrox/rox/pkg/utils"
 )
 
 // CurrentDBVersionSeqNum is the current DB version number.
 // This must be incremented every time we write a migration.
 // It is a shared constant between central and the migrator binary.
 func CurrentDBVersionSeqNum() int {
-	// If PostgresDatastore is enabled, we consider all migrations to Postgres are following
-	// the current migrations.
-	// if features.PostgresDatastore.Enabled() {
-	//	return internal.CurrentDBVersionSeqNum + internal.PostgresDBVersionPlus
-	// }
+	if features.PostgresDatastore.Enabled() {
+		return internal.CurrentDBVersionSeqNum + internal.PostgresDBVersionPlus
+	}
```

Comment:
I'm curious about a scenario. 
CurrentDBVersionsSeqNum = 107
PostgresDBVersionPlus = 56

So we upgrade from Rocks to Postgres.  Migrations will run and get Rocks to level 107.  The migration loop will go until it hits 163.  Starting at 108 it will run the n_X_to_n_Y migrations.  When all that is done the sequence number stored in the postgres version will be 163.  But what happens if we need to add a new migration after Postgres is deployed while still supporting RocksDB?  

The way these numbers work, we would create m_107_m_108 that would handle the Rocks->Rocks migration.  
Then we would have an n_57_to_n_58 migration that would handle Rocks -> Postgres.
Where would the Postgres->Postgres migration fall?  We couldn't do it at 108 because a Postgres upgrade would never see it as Postgres would think it is at 163.  Would we want to do that in the n_57_to_n_58 migration?  I'm curious what that scenario looks like from a numbering scheme and what goes where.

Reply:
Yes, what you thought is correct.

After we determine the version to release, new migrations would increase the PostgresDBVersionPlus only. So, this process is going to be three phrases:
1. Now till we are working on the first Postgres release, we increase CurrentDBVersionsSeqNum to add new migrations so we can release them before all legacy->Postgres migration.
2. When we work on the Postgres version to release, we add all migrations that should happen after Postgres migration with PostgresDBVersionPlus.
3. After Postgres version is releases or we are working on the postgres release +1, we merge this two sequence. We do not have to merge the directory names.

Reply:
Are you saying all Postgres migrations go in the new directory structure?  Is this written down somewhere?  I took a look at your migration design doc and it didn't go in detail about sequence number and directory structure.  I think we should document it some place if it isn't.

Reply:
Yes, I will update the doc.